app/src/lib/editors/win32.ts

```diff
@@ -10,663 +10,592 @@ import {
-      )
-    case ExternalEditor.SublimeText:
-      return (
-        displayName === 'Sublime Text' && publisher === 'Sublime HQ Pty Ltd'
-      )
-    case ExternalEditor.CFBuilder:
-      return (
-        (displayName === 'Adobe ColdFusion Builder 3' ||
-          displayName === 'Adobe ColdFusion Builder 2016') &&
-        publisher === 'Adobe Systems Incorporated'
-      )
-    case ExternalEditor.Typora:
-      return displayName.startsWith('Typora') && publisher === 'typora.io'
-    case ExternalEditor.SlickEdit:
-      return (
-        displayName.startsWith('SlickEdit') && publisher === 'SlickEdit Inc.'
-      )
-    case ExternalEditor.Webstorm:
-      return (
-        displayName.startsWith('WebStorm') && publisher === 'JetBrains s.r.o.'
-      )
-    case ExternalEditor.Phpstorm:
-      return (
-        displayName.startsWith('PhpStorm') && publisher === 'JetBrains s.r.o.'
-      )
-    case ExternalEditor.NotepadPlusPlus:
-      return (
-        displayName.startsWith('Notepad++') && publisher === 'Notepad++ Team'
-      )
-    case ExternalEditor.Rider:
-      return (
-        displayName.startsWith('JetBrains Rider') &&
-        publisher === 'JetBrains s.r.o.'
-      )
-    default:
-      return assertNever(editor, `Unknown external editor: ${editor}`)
-  }
-}
-
-function getKeyOrEmpty(
-  keys: ReadonlyArray<RegistryValue>,
-  key: string
-): string {
-  const entry = keys.find(k => k.name === key)
-  return entry && entry.type === RegistryValueType.REG_SZ ? entry.data : ''
+) => boolean
+
+/** Represents an external editor on Windows */
+interface IWindowsExternalEditor {
+  /** Name of the editor. It will be used both as identifier and user-facing. */
+  readonly name: string
+
+  /**
+   * Set of registry keys associated with the installed application.
+   *
+   * Some tools (like VSCode) may support a 64-bit or 32-bit version of the
+   * tool - we should use whichever they have installed.
+   */
+  readonly registryKeys: ReadonlyArray<{ key: HKEY; subKey: string }>
+
+  /**
+   * List of path components from the editor's instalation folder to the
+   * executable shim.
+   **/
+  readonly executableShimPath: ReadonlyArray<string>
+
+  /**
+   * Whether the or not the provided executable for the editor needs to be run
+   * inside of a shell.
+   */
+  readonly usesShell?: boolean
+
+  /**
+   * Function that maps the registry information to a list of known installer
+   * fields (display name, publisher and installation path).
+   *
+   * Receives the collection of registry key-value pairs for the app.
+   */
+  readonly appInformationExtractor?: AppInformationExtractor
+
+  /**
+   * Function to check if the found installation matches the expected identifier
+   * details.
+   *
+   * @param displayName The display name as listed in the registry
+   * @param publisher The publisher who created the installer
+   */
+  readonly expectedInstallationChecker: ExpectedInstallationChecker
```

Comment:
I'm not particularly proud of these properties or their names or anything, but I'm open to suggestions 😂 

This felt like the best way to avoid scattering specific editor stuff all over this file 🤔

Reply:
I'm fine with these names but I would probably have called the `appInformationExtractor` method something like `getAppInfo` or `getUninstallInfo`. Your call tho